network/network.go

```diff
@@ -503,13 +504,14 @@ func (n *network) AllowConnection(nodeID ids.NodeID) bool {
 		return true
 	}
 	_, areWeAPrimaryNetworkAValidator := n.config.Validators.GetValidator(constants.PrimaryNetworkID, n.config.MyNodeID)
-	return areWeAPrimaryNetworkAValidator || n.ipTracker.WantsConnection(nodeID)
+	return areWeAPrimaryNetworkAValidator || n.ipTracker.WantsConnection(nodeID) || n.config.ConnectToAllValidators
```

Comment:
Why do we check the `ConnectToAllValidators` config here? I thought this was checked in `WantsConnection` and if this is a non-validator we're connecting to i think we'll blindly accept the connection even if we're also a non-validator.